lib/Db/TagMapper.php

```diff
@@ -159,21 +126,24 @@ public function untagMessage(Tag $tag, string $messageId): void {
 
 	/**
 	 * @param Message[] $messages
+	 * @param string $userId
 	 * @return Tag[][]
 	 */
-	public function getAllTagsForMessages(array $messages): array {
+	public function getAllTagsForMessages(array $messages, string $userId): array {
 		$ids = array_map(function (Message $message) {
 			return $message->getMessageId();
 		}, $messages);
 
 		$qb = $this->db->getQueryBuilder();
-		$tagsQuery = $qb->select('t.*', 'mt.imap_message_id')
+		$tagsQuery = $qb->selectDistinct(['t.*', 'mt.imap_message_id'])
 			->from($this->getTableName(), 't')
-			->join('t', 'mail_message_tags', 'mt', $qb->expr()->eq('t.id', 'mt.tag_id', IQueryBuilder::PARAM_INT))
+			->join('t','mail_message_tags', 'mt',$qb->expr()->eq('t.id', 'mt.tag_id'))
```

Comment:
spaces and parameter argument can be reverted to the old version, right?

Reply:
yes